Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+        event.stopPropagation();
+        isSelectionUpdated = true;
+        oldIndex = currentItemIndex;
+        while (true) // skip separators
+        {
+          if (currentItemIndex < itemInfos.length - 1)
+          {
+            currentItemIndex++;
+            if (currentItemIndex === 1 && oldIndex === -1)
+              oldIndex = 0;
+          }
+          else
+          {
+            currentItemIndex = 0;
+          }
+
+          if (oldIndex === currentItemIndex)
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0) // skip separators
+            break;
+        }
+        break;
+      case 38: // up arrow
+        event.preventDefault();
+        event.stopPropagation();
+        isSelectionUpdated = true;
+        oldIndex = currentItemIndex >= 0 ? currentItemIndex : itemInfos.length - 1;
+        while (true) // skip separators
+        {
+          if (currentItemIndex > 0)
+            currentItemIndex--;
+          else
+            currentItemIndex = itemInfos.length - 1;
+
+          if (oldIndex === currentItemIndex)
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0)
+            break;
+        }
+        break;
+    }
+
+    if (isSelectionUpdated)
+    {
+      $("li", DropDownMenu._currentPopup).removeClass (DropDownMenu._itemSelectedClassName);
+      if (currentItemIndex >= 0 && currentItemIndex < itemInfos.length)
+      {
+        var dropDownMenuItem = $(dropDownMenuItems[currentItemIndex]);
+        if (dropDownMenuItem.children('a').length > 0) // skip separators
+        {
+          dropDownMenuItem.addClass(DropDownMenu._itemSelectedClassName);
+          var nonJQueryAnchor = dropDownMenuItem.children ('a')[0];
+          // do not use jQuery focus() here because jQuery causes the focus/blur events to fire in the wrong order (first focus, then blur)
+          nonJQueryAnchor.focus();
+        }
+      }
+    }
+  }
+
+  private static GetItemEnabled (itemInfo: DropDownMenu_ItemInfo, selectionCount: number): boolean
+  {
+    var isEnabled = true;
+    if (itemInfo.IsDisabled)
+    {
+      isEnabled = false;
+    }
+    else
+    {
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionExactlyOne
+          && selectionCount != 1)
+      {
+        isEnabled = false;
+      }
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionOneOrMore
+          && selectionCount < 1)
+      {
+        isEnabled = false;
+      }
+    }
+    return isEnabled;
+  }
+
+  private static GetTarget (event: JQueryMouseEventObject, tagName: string): Element
+  {
+    var element: Nullable<Element> = event.target;
+    while (element && element.tagName != tagName)
+      element = element.parentNode as Nullable<Element>;
+    // more fun with IE, sometimes event.target is empty, just ignore it then
+    if (!element) // TODO: What is this? This should also fail by the caller.
```

Comment:
I don't understand the question. See line 881.

Reply:
Why is `[]` returned if `!element`?

Reply:
ah, yes. I think this acted as a null-object for this particular scenario, allwing the callsite to just do NOPs in the IF-case. Sicne it's IE code and IE is dead, it's not relevant either way.